install-deps.sh

```diff
@@ -146,19 +146,49 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+
+function clean_boost_on_ubuntu {
+    ci_debug "Running clean_boost_on_ubuntu() in install-deps.sh"
+    # Find currently installed version. This may not work if there are
+    # multiple versions installed.
+    local installed_ver=$(apt -qq list --installed ceph-libboost*-dev 2>/dev/null |
+                              cut -d' ' -f2 |
+                              cut -d'.' -f1,2 | sort | uniq)
+    # If installed_ver contains whitespace, we can't really count on it,
+    # but otherwise, bail out if the version installed is the version
+    # we want.
+    if test -n "$installed_ver" && echo "$installed_ver" | grep -v -q '[[:space:]]'; then
```

Comment:
I don't think this works as expected.  First, `echo` appends a newline, so there is always going to be a newline there even with a single version.  Second, `grep` is line-based, so it wouldn't match newlines unless special care is taken (like -z/-Z options of GNU grep).

Please test your changes at least on some toy examples before pushing -- I have lost count of the number of silly bugs like this found in this function in review.  For a proper test, I would suggest setting up some Ubuntu containers, adding those shaman repos and running `install-deps.sh` through scenarios:
- no boost packages installed
- one version installed, matches the desired version
- one version installed, doesn't match the desired version
- two versions installed, one matches the desired version
- two versions installed, none match the desired version